t/exception.t

```diff
@@ -1,8 +1,55 @@
 #!/usr/bin/env perl
 
+use warnings;
 use strict;
 use lib qw(../lib lib ../t t);
-use TestUtil;
-use Test::More  tests => 1;
+use Test::More;
+use Test::Exception;
+use English qw(-no_match_vars);
 
-require_ok( 'Workflow::Exception' );
+use_ok( 'Workflow::Exception', qw(workflow_error validation_error condition_error configuration_error persist_error) );
+
+{
+    throws_ok {
+        workflow_error('test ', 'workflow_error')
+    } 'Workflow::Exception', 'workflow workflow_error exception';
+    like($EVAL_ERROR, qr/test workflow_error/, "expected error string: $EVAL_ERROR");
+}
+
+{
+    throws_ok {
+        Workflow::Exception->throw(
+            message => 'workflow exception',
+            foo     => 'bar',
+        );
+    } qr/messageworkflow exceptionfoobar/, 'exception thrown';
+}
+
+{
+    throws_ok {
+        validation_error('test ', 'validation_error', { foo => 'bar' })
+    } 'Exception::Class::Base', 'workflow validation_error exception';
+    like($EVAL_ERROR, qr/unknown field foo passed to constructor for class Workflow::Exception::Validation/, "expected error string: $EVAL_ERROR");
+}
+
+{
+    throws_ok {
+        condition_error('test ', 'condition_error')
+    } 'Workflow::Exception', 'workflow condition_error exception';
+    like($EVAL_ERROR, qr/test condition_error/, "expected error string: $EVAL_ERROR");
```

Comment:
I don't understand the like tests-description you specify: like($EVAL_ERROR, qr/test condition_error/, "expected error string: $EVAL_ERROR"); which seems to suggest that you were expecting `$EVAL_ERROR`'s content as the error, but you're testing for the existence of `test condition_error` in the output?

Reply:
You are right, the message is communicating nonsense